### ISSUE TITLE
Installation causes external requests

### ISSUE THREAD

**Issue by mtompkins** [thorsten/phpMyFAQ#1067]
It seems some component of phpMyFAQ is causing an external request from 81.246.43.103.
In short, my use of phpMyFAQ is purely internal and I have safeguards in place to ensure that is the case. There should be no 'announcement' or other information from the internal network disclosed to external entities. There should be no reason for an external to be requesting anything within my DMZ zone.

I find that after installing phpMyFAQ, requests are made to apache host from the above identified IP address, which resolves to the following information: http://www.networksolutions.com/whois/results.jsp?ip=81.246.43.103

There is no DNS registration or other CNAME publicly registered for the virtual server I am using.
This condition has been replicated on an independent network with the same results.

I find this quite troubling. 
Luckily I control access at the global level of apache which is preventing this access.

Here is the apache log:

[Fri Nov 27 11:09:39.111335 2015] [access_compat:error] [pid 1839] [client 81.246.43.103:53407] AH01797: client denied by server configuration: /var/www/faq/
[Fri Nov 27 11:09:39.112665 2015] [access_compat:error] [pid 1839] [client 81.246.43.103:53407] AH01797: client denied by server configuration: /var/www/custom-errors/401error.html

Here is the associated access attempt: 
81.246.43.103 - - [27/Nov/2015:11:09:39 -0500] "GET / HTTP/1.1" 401 28705 "-" "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36"

**Comment by thorsten** [comment]
How did you install phpMyFAQ? Which version?

**Comment by thorsten** [comment]
I do not find this IP in the source code, downloaded from phpmyfaq.de

**Comment by winks** [comment]
Hi, could you please doublecheck in a private browsing session with all addons disabled that it's not some browser extension?

**Comment by mtompkins** [comment]
@winks This is from the apache log, not from my internal browser session. It was captured overnight. There have been others. I run fail2ban (in addition to other safeguards) so that once it is identified it is banned.

@thorsten I agree there isn't a source of that IP from the code within the repo. Unfortunately (and not surprisingly) the IP address changes over time. It is very likely this will be a hard one to hunt down as there are so many external modules used in your offering. I do not think this is the direct fault of your code per se, but much more likely a consequence of one of your modules.

The install I have (as well as the external test I asked a friend to do) was direct from your site link (not built, regular install which is just a file copy).

**Comment by thorsten** [comment]
Is your installation publicly available?

**Comment by mtompkins** [comment]
For clarity, here is another capture from a few days ago:
[Sun Nov 15 17:39:03.822999 2015] [access_compat:error] [pid 2994] [client 45.55.79.204:41824] AH01797: client denied by server configuration: /var/www/faq/
[Sun Nov 15 17:39:03.823511 2015] [access_compat:error] [pid 2994] [client 45.55.79.204:41824] AH01797: client denied by server configuration: /var/www/custom-errors/401error.html

I waited until I noticed a few of these before reporting.

**Comment by mtompkins** [comment]
@thorsten "Is your installation publicly available?" - as indicated, no which is why this is such a concern.

**Comment by thorsten** [comment]
Maybe it's a captured Windows PC crawling randomly server IPs?

**Comment by mtompkins** [comment]
A server IP wouldn't resolve to that directory. My host has a default directory to catch IP crawls.

Also FWIW I do not have twitter or any other external integration setup.

**Comment by thorsten** [comment]
Well, it's hard to see what's happening from outside...

**Comment by mtompkins** [comment]
The steps to reproduce on a VM are fairly straight forward:

Set up an apache2 VM (e.g. Debian / Ubuntu, etc.)
Install phpMyFAQ
Create a virtual host limiting the access to local LAN:
example (change IP scope as necessary):

    DocumentRoot "/var/www/faq"

	CustomLog ${APACHE_LOG_DIR}/access.log combined

    <Location />
    Order allow,deny    
	Allow from 127.0.0.1 192.168.1    
	Allow from ::1    
	Satisfy Any   
    </Location>

Then monitor for unauthorized access.

**Comment by mtompkins** [comment]
There is one item in play that might be contributing and I've sent in a request for information. The site does have a registered SSL with StartSSL.com. That is the only external party who has knowledge of the URL (as they must have) in order to issue the certificate.

Perhaps somehow that is otherwise leaking to the internet. I will advise when I hear their response.

**Comment by mtompkins** [comment]
I exchanged emails with the COO / CTO of StartSSL. Their position is understandable - and in short although they don't say that it is impossible to be them (they've made a great case on how their systems don't know the IP address at any time) they believe the greatest probability is that the cause is somewhere within the open source phpMyFAQ solution. I tend to agree.

Again I recognize that this is a nightmare to track down - but as this is a real security concern it seemed necessary to share my discovery. Since my last post there has been another attempt, this time trying to access an asterisk backend. I presume my URL and IP has made it on to some list somewhere. I plan on asking the ISP to issue my a new lease. That should give us an idea how bad this condition might be.

**Comment by thorsten** [comment]
Is possible to analyze the network traffic?

**Comment by mtompkins** [comment]
Inbound my system immediately bans the source of the incoming request. As far as outbound - I have monitored for about a month and haven't been able to find anything that looks relevant. The only thing I can search packets for is the URL - and there is nothing to guarantee that it is being passed in clear text. In addition, it may be so infrequent that it doesn't happen often enough to spot.

**Comment by thorsten** [comment]
Any news on that?

**Comment by mtompkins** [comment]
Still exists. I have mediated it within my own infrastructure but it occurs on any new installation.

**Comment by thorsten** [comment]
Always the same IP?

**Comment by mtompkins** [comment]
No - source is global and various IPs. "Feels" like some automation of course. All I can really say is that it is always the URL that serves up phpmyfaq.

**Comment by mtompkins** [comment]
... I should clarify. It makes the request with the phpmyfaq URL at the correct public IP address. This IP address is not publicized on external DNS servers so it is not doing a public internet DNS lookup to resolve the IP. I captured  inbound request headers to determine this.

**Comment by thorsten** [comment]
Could it be a Apache misconfiguration? https://www.apachelounge.com/viewtopic.php?p=28631

**Comment by mtompkins** [comment]
Not in this instance, no.

**Comment by rtreffer** [comment]
Could you verify that the HOST header is actually needed? Apache, if configured with multiple named hosts will pick one as the "default" vhost to server requests without a HOST header.

```
$ apachectl -S
VirtualHost configuration:                                                                                                           
*:80                   is a NameVirtualHost                                                                                          
         default server domain.under.your.tld (/relevant/config/file:line)
```

This would lift the "CNAME is not know" constraint. (Anything that does not have a HOST header or a HOST header that does not resolve is routed to the default server - even if that default server has a different name!)

Do you enforce SSL for everyone? You might otherwise leak the URL through HTTP referer. (The company owning the IP looks rather strange and might have done a referer analysis).

The user agent points to Chrome on Windows. Bots and alike usually pick IE on Windows if they fake anything. There is a very slight chance that someone opened his browser and accessed the site (causing a 401). You might want to log
- Any cookies
- X-FORWARDED-FOR header
- HOST header (just to be sure)
- Anything that would indicate if it was a HTTP or HTTPs request
This would explain why it can work without a CNAME. This could provide some nice hints for debugging.

You should also double check from external that the name does not resolve at all. There is a slight chance that another star record catches your request.

**Comment by mtompkins** [comment]
**UPDATE**: So after months of monitoring and picking this apart I finally have some additional information. Having exhausted all things 'software', on a hunch I replaced Apache with Nginx, revoked all active certificates used for phpmyfaq, changed the internal URL and had new certificates issued to match the changes (same Cert-Auth: StartSSL). I then started to monitor after these changes. Going on 45 days and no external requests to this new URL.

While I'm not sure where the culprit is yet, I'm more and more confident that it isn't originating from phpmyfaq. While I'd like to make a statement that I'm positive, I'm not quite there yet.  However, basis these changes and the lack of new external requests, I think in fairness I should lean toward closing this ticket until I have better evidence that it is the phpmyfaq code.

Appreciate the comments from those above and especially @thorsten
